build-system/eslint-rules/html-template.js

```diff
@@ -159,32 +147,146 @@ module.exports = function (context) {
     return matches;
   }
 
-  return {
-    CallExpression(node) {
-      if (/test-/.test(context.getFilename())) {
-        return;
-      }
+  function cannotImplicitlyReturnCalls(node) {
+    context.report({
+      node,
+      message: `Do not implicitly return ${node.tag.callee.name}()\`...\`, since it requires a helper constant in function scope.`,
+      fix: function* (fixer) {
+        yield fixer.insertTextBefore(node, `{\nreturn (`);
+        yield fixer.insertTextAfter(node, `);\n}`);
+      },
+    });
+  }
 
-      const {callee} = node;
-      if (callee.type !== 'Identifier') {
-        return;
-      }
+  const checkedScopeBlocks = new Set();
+  function tagMustBeHelperConstant(node) {
```

Comment:
Curious: Will this rule prevent someone from doing something like this? (Using `foo` instead of the `html` keyword.)

```js
const foo = htmlFor(this.doc_);
bar = foo` <div class="some-tag"></div> `;
```

Sorry if this is already documented somewhere.

Reply:
No problem! This works the same as previously.

This fails because `htmlFor(...)` should be assigned to `html`:
```js
// htmlFor result must be stored into a helper constant named "html". (at 1:13)
   const foo = htmlFor(this.doc_);
// ------------^
```
This doesn't match because we don't know about template literal tag `foo`:
```js
bar = foo` <div class="some-tag"></div> `;
```

Reply:
Very nice!